### ISSUE TITLE
How to stub exception from async method since NNBD ?

### ISSUE THREAD

**Issue by woprandi** [dart-lang/mockito#358]
I often do that : 
```dart
when(api.patchObject(any)).thenAnswer((_) async => throw Exception());
```
To test my app's behavior on error. I'm working on the NNBD migration and now I get this error : 
```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following _TypeError was thrown running a test:
type 'Null' is not a subtype of type 'Future<Never>'

When the exception was thrown, this was the stack:
#5      FakeAsync.flushMicrotasks (package:fake_async/fake_async.dart:193:32)
#6      FakeAsync._fireTimersWhile (package:fake_async/fake_async.dart:232:5)
#7      FakeAsync.elapse (package:fake_async/fake_async.dart:137:5)
#8      AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:960:28)
#11     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#12     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:956:27)
#13     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:640:23)
#14     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:634:38)
#17     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#18     WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:634:27)
#19     main.<anonymous closure> (file:///home/william/Documents/Otera/Repos/planning_front/test/main_page_test.dart:301:18)
<asynchronous suspension>
<asynchronous suspension>
(elided 10 frames from dart:async and package:stack_trace)
```

I didn't find any information about NNBD migration for this specific use-case

**Comment by srawlins** [comment]
The [NULL_SAFETY_README.md](https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md) should help.

**Comment by woprandi** [comment]
I understood I have only this issue when my method (`patchObject` here) return `Future<Something>` and not with `Future<void>` 
Am I supposed to do that : https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md#manually-override-a-method-with-a-non-nullable-return-type ?

**Comment by srawlins** [comment]
That's correct. `null` is a valid value for `Future<void>`, but not for `Future<SomethingNonNullable>`.

**Comment by woprandi** [comment]
I tested my code with a real scenario, I also have this exception even if it's not blocking. The `catchError` block is still executed. 
I found 2 ways to fix it : 
Handle error synchronously (try/catch + await) or add an empty then block

```dart
future
.then((_){})
.catchError((_) {...})
```

The issue seems not about mocking but the regression with NNBD and `future.catchError((){})` statement. I didn't find any warning about that in the NNBD migration.
